### DISCUSSION TITLE
Creating a temporary access token via API for a private github repo

### DISCUSSION THREAD

**Discussion by suntong** [community/community#49725] [Apps, API and Webhooks]
### Select Topic Area

General

### Body

I want to temporary give access to view/download a single file from my private repository using **url with token for single file** form. 

- This is generated by github automatically when we click the 'raw' button on the GH web site. 
- How to acquire such single file token myself via some sorts of API call?

Thanks

Full quote from [here](https://stackoverflow.com/a/65367523/2125837)

> I think it is a little bit dangerous and not good way to issue a personal access token which can access all repositories even just for download a single file from my private repository. 
> 
> ## How to -
> I would love to recommend using **url with token for single file**. Don't worry. The token string will generated by github automatically. You can get this url on your source code page.
> 
> 1. Go to source code page what you want to download via curl or wget, etc
> 2. Find 'raw' button and click it.
[![enter image description here][1]][1]
> 3. New page opened and just copy the url. This url look like below:  
  (https://raw.githubusercontent.com/USERNAME/REPONAME/BRANCHNAME/FILENAME?token=TOKENSTRING). 
> 4. You can download a file using this url

  [1]: https://i.stack.imgur.com/XPvRl.png

**Comment by Aadhavancnp** [comment]
Refer to this Article in StackOverflow https://stackoverflow.com/questions/31264165/can-i-share-my-private-github-repository-by-link
It may Help!!

**Reply by suntong** [reply]
Thanks for the suggestions of doing it _manually_.
